### DISCUSSION TITLE
3rd party libraries with Swift break, because `React-RCTFabric`, `React-Codegen`, `RCT-Folly`, ... do not define modules

### DISCUSSION THREAD

**Discussion by mateusz1913** [reactwg/react-native-new-architecture#34] [Libraries]
I recently tried to play with creating a simple POC library, that is compatible with "New Architecture", and uses Swift on the iOS side, in order to learn how to migrate my library [react-native-avoid-softinput](https://github.com/mateusz1913/react-native-avoid-softinput).

When I added Swift files (inside the library directory) and made Pods synchronization in the example project, it failed with:
```
[!] The following Swift pods cannot yet be integrated as static libraries:

The Swift pod `poc-library` depends upon `React-RCTFabric`, `React-Codegen`, `RCT-Folly`, `RCTTypeSafety`, and `ReactCommon`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```

I found that it was already highlighted by @mrousavy [in Playbook PR](https://github.com/facebook/react-native-website/pull/2879#discussion_r767871881).

This is a blocker for migration of react-native-avoid-softinput and other libraries that are written in Swift.

Is there some roadmap for ~~support of Swift libraries with "New Architecture"~~ those pods (`RCT-Folly`, `ReactCommon`, etc.) to be compatible with Swift code?

**Comment by sammy-SC** [comment]
Hello @mateusz1913,

The new architecture is written in C++ and for React Native to support Swift out of the box, Swift needs to work with C++. At this moment, C++ interoperability is an ongoing effort within Swift project. There seems to be a lot of interest in the space and [Swift has been making progress](https://forums.swift.org/t/swift-and-c-interoperability-workgroup-announcement/54998) towards the interoperability. When Swift can work with C++, it should be possible to build native components in Swift as the new architecture's API are in C++.

For now, you should be able to wrap your Swift component in Obj-C++ native component. It is an extra step where C++ APIs will need to be manually transformed to Swift, but it should work.

**Reply by mateusz1913** [reply]
Hi @sammy-SC , thx for your response, but this is not what I actually meant to ask for. I am aware of Swift-C++ interoperability status. The problem is different
- libs depending on "New Architecture" Pods (e.g. `React-RCTFabric`, `RCT-Folly`) break during `pod install`, because those pods are not compatible with Swift code (`[!] The following Swift pods cannot yet be integrated as static libraries`)

I will change title and update description, to highlight react pods being incompatible and not C++ with Swift

**Comment by mateusz1913** [comment]
It will probably be resolved with https://github.com/facebook/react-native/commit/709a459b1ef7968d447c0a5dbe5cd44d1e03fcb4

I will mention it in 0.68 & 0.69 releases discussions
